src/modeler/AutoRest.Swagger.Tests/SwaggerModelerValidationTests.cs

```diff
@@ -42,253 +42,250 @@ internal static void AssertOnlyValidationMessage(this IEnumerable<ValidationMess
     [Collection("Validation Tests")]
     public partial class SwaggerModelerValidationTests
     {
-        private IEnumerable<ValidationMessage> ValidateSwagger(string input, ServiceDefinitionDocumentType serviceDefDocType = ServiceDefinitionDocumentType.ARM, 
-            ServiceDefinitionDocumentState mergeState = ServiceDefinitionDocumentState.Composed)
+        private IEnumerable<ValidationMessage> ValidateSwagger(string input, ServiceDefinitionMetadata metadata)
         {
-
             // Most rules are to be applied for ARM documents
             // Also, most rules need to be run over the composite document (i.e. AFTER merge state)
             // hence the defaults
             using (NewContext)
             {
                 var validator = new RecursiveObjectValidator(PropertyNameResolver.JsonName);
                 var serviceDefinition = SwaggerParser.Parse(input, File.ReadAllText(input));
-
-                var metaData = new ServiceDefinitionMetadata
-                {
-                    ServiceDefinitionDocumentType = serviceDefDocType,
-                    MergeState = mergeState
-                };
-                
-                return validator.GetValidationExceptions(new Uri(input, UriKind.RelativeOrAbsolute), serviceDefinition, metaData).OfType<ValidationMessage>();
+                return validator.GetValidationExceptions(new Uri(input, UriKind.RelativeOrAbsolute), serviceDefinition, metadata).OfType<ValidationMessage>();
             }
         }
 
+        private ServiceDefinitionMetadata GetMetadataForRuleTest(Type rType)
+        {
+            var rule = (Rule)Activator.CreateInstance(rType);
+            return new ServiceDefinitionMetadata
+            {
+                ServiceDefinitionDocumentType = rule.ServiceDefinitionDocumentType,
+                MergeState = rule.ValidationRuleMergeState
+            };
+        }
+
         [Fact]
         public void MissingDescriptionValidation()
         {
```

Comment:
Thank you! It took me a good amount of time trying to figure out the best way to generalize this further.